### DISCUSSION TITLE
[Language Detection] Whisper Large-v3 cannot detect cantonese audio as "yue" language

### DISCUSSION THREAD

**Discussion by josephwong14wkh** [openai/whisper#2419] [Q&A]
I am using whisper large-v3 to transcribe an audio spoken in Cantonese. However, I found that whisper detected the language as "zh" (Chinese), instead of "yue" (Cantonese), and the transcription becomes written-language format, instead of spoken-language format. Is this behavior normal? Is there any method to solve this issue?

**Comment by samarasimhapeyala** [comment]
Can you try passing the language(of audio) argument as well `--language yue` or `--language cantonese`

**Comment by josephwong14wkh** [comment]
Thank you for your response. When I pass the language argument `--language yue`, the text is transcribed in written-language format, rather than spoken-language format. But in my case, the transcription is not very important. I just want whisper to differentiate whether the audio is spoken in mandarin or cantonese. However, it seems what whisper cannot distinguish these two languages (both large-v3 and large-v3 turbo version). Is there any idea for language detection part?

**Comment by windring** [comment]
same problem

**Comment by bmkor** [comment]
Yes, encountered the same problem when trying to solve the issue as discussed [here](https://huggingface.co/datasets/amphion/Emilia-Dataset/discussions/7)

**Comment by elleryleung-setsail** [comment]
@bmkor I got the same problem and yet when I am using `large-v3` and it still can't detect the `yue` language.  Any recommendation?

Thank you.
